3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,159 @@
+  };
+
+  function calculateTime (travelInformation){
+    const { speed, destinationDistance } = travelInformation;
+    
+    const totalTimeInHours = destinationDistance / speed;
+
+    const hours = Math.floor(totalTimeInHours);
+    const minutes = Math.round((totalTimeInHours - hours) * 60);
+
+    const formattedTime = `${hours} hours and ${minutes} minutes`;
+
+    return formattedTime;
+
+  }
+  
+  const travelTime = calculateTime(travelInformation);
+  console.log(travelTime); 
+
+  //Series duration of my life
+
+  const seriesDurations = [
+    {
+      title: "Game of thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+  ];
+  
+  function calculateMinutes(seriesDurations) {
+    let totalminutes = 0;
+
+    for ( const series of seriesDurations){
+      const {days, hours, minutes} = series;
+      totalminutes = days * 24 * 60 + hours * 60 + minutes; 
+    }
+    return totalminutes;
+  }
+
+    function calculatePercentageOfLife(averageLife , totalminutes){
+      const minutesInYear = 365 * 24 * 60;
+      const percentage = (totalminutes / (averageLife * totalminutes) * 100 );
+      return percentage.toFixed(3);
+    }
+
+    function logOutSeriesText (){
+      const minutesWatchedSeries = calculateMinutes(seriesDurations);
+      const averageLife = 80;
+
+      let seriesOutput = "";
+
+      for (const series of seriesDurations){
+        const {title, days, hours, minutes} = series;
+        const seriesMinutes = days * 24 * 60 + hours * 60 + minutes; 
+        const percentageOfLife = calculatePercentageOfLife(seriesMinutes, averageLife)
+        seriesOutput += `${title} took ${percentageOfLife} in my life\n`
+
+      }
+
+      const totalPercentage = calculatePercentageOfLife(minutesWatchedSeries, averageLife);
+
+      console.log(seriesOutput);
+      console.log(`Games of Throne took ${totalPercentage}% of my life`)
+
+    }
+  
+  logOutSeriesText();
+
+//NOnoN0nOYes (Note taking app)
+
+//Step 1
+
+const notes = [];
+
+function saveNote(content, id) {
+ const note = { content , id}
+  notes.push(note);
+
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); 
+
+//Step 2
+
+function getNote(id) {
+  const foundNotes = notes.find((note)=>note.id === id)
+  
+  if(foundNotes){
+    return foundNotes
+  } else {
+    return null;
+  }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+//Step 3
+
+function logOutNotesFormatted() {
+  if (notes.length === 0){
+    console.log("Not found")
+  } else {
+    console.log( "All notes are here")
+    notes.forEach((notes) => {
+      console.log(` ${notes.id}: ${notes.content}`)
+    })
+  }
+}
+
+logOutNotesFormatted(); // should log out the text below
+
```

Comment:
🎉 Great job! Your code accomplishes the objectives of the exercise effectively and uses some good practices like the usage of find method for arrays and the concise ES6 syntax for object property assignments.

👍 I like your approach to checking if the notes array is empty before attempting to print them.

✨ Suggestions:

In your `getNote` function, the variable name `foundNotes` suggests multiple notes but the function is designed to return only one. Consider renaming it to `foundNote`.

For the error handling in `getNote`, you could consider providing a more descriptive error message if a note is not found by id or if an invalid id is provided.

Inside the forEach callback in `logOutNotesFormatted`, the argument name `notes` is misleading since it represents a single `note`. Consider renaming it to `note`.

You might want to include error handling for situations where the same note ID is used twice, to ensure data integrity.